src/utilities/sortIntrospection.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {
+  IntrospectionQuery,
+  IntrospectionType,
+  IntrospectionField,
+  IntrospectionDirective,
+} from './introspectionQuery';
+
+/**
+ * Sort the result of Introspection Query.
+ */
+export function sortIntrospectionQuery(
+  introspection: IntrospectionQuery,
+): IntrospectionQuery {
+  const schema = introspection.__schema;
+  return {
+    __schema: {
+      queryType: schema.queryType,
+      mutationType: schema.mutationType,
+      subscriptionType: schema.subscriptionType,
+      types: sortByName(schema.types).map(sortType),
+      directives: sortByName(schema.directives).map(sortDirective),
+    },
+  };
+}
+
+function sortDirective(
+  directive: IntrospectionDirective,
+): IntrospectionDirective {
+  return {
+    name: directive.name,
+    description: directive.description,
+    locations: sortBy(directive.locations, x => x),
```

Comment:
there's no reason to use `sortBy` here: just `directive.locations.slice().sort((a, b) => a.compareLocale(b));`

This is useful because then we can make `sortBy` take a mutable array, and avoid one array-copy-per-`sortByName`